linkml/generators/pydanticgen/array.py

```diff
@@ -175,6 +176,124 @@ def _validate(_v: Union[List[_T], list]):
+            )
+
+    @staticmethod
+    def array_consistent_n_dimensions(array: ArrayExpression):
+        """
+        Complex arrays with both exact/min/max_number_dimensions and parameterized dimensions
+        need to have the exact/min/max_number_dimensions greater than the number of parameterized dimensions!
+        """
+        if not array.dimensions:
+            return
+
+        for field_name in _BOUNDED_ARRAY_FIELDS:
+            field = getattr(array, field_name, None)
+            if field and field < len(array.dimensions):
+                raise ValidationError(
+                    "if exact/minimum/maximum_number_dimensions is provided, "
+                    "it must be greater than the parameterized dimensions. "
+                    f"got\n- {field_name}: {field}\n- dimensions: {array.dimensions}"
+                )
+
+    @staticmethod
+    def array_dimensions_ordinal(array: ArrayExpression):
+        """
+        minimum_number_dimensions needs to be less than maximum_number_dimensions when both are set
+        """
+        if array.minimum_number_dimensions is not None and array.maximum_number_dimensions:
+            if array.minimum_number_dimensions > array.maximum_number_dimensions:
+                raise ValidationError(
+                    "minimum_number_dimensions must be lesser than maximum_number_dimensions when both are set. "
+                    f"got minimum: {array.minimum_number_dimensions}, maximum: {array.maximum_number_dimensions}"
+                )
+
+    @staticmethod
+    def array_explicitly_unbounded(array: ArrayExpression):
+        """
+        Complex arrays with a minimum_number_dimensions and parameterized dimensions
+        need to either use exact_number_dimensions to specify extra anonymous dimensions
+        or set maximum_number_dimensions to ``False`` to specify unbounded extra anonymous
+        dimensions to avoid ambiguity.
+        """
+        if array.minimum_number_dimensions is not None and array.maximum_number_dimensions is None and array.dimensions:
+            raise ValidationError(
+                (
+                    "Cannot specify a minimum_number_dimensions while maximum is None while using labeled dimensions - "
+                    "either use exact_number_dimensions > len(dimensions) for extra parameterized dimensions or set "
+                    "maximum_number_dimensions explicitly to False for unbounded dimensions"
+                )
+            )
+
+    @staticmethod
+    def dimension_exact_cardinality(dimension: DimensionExpression):
+        """Dimensions can only have exact_cardinality OR min/max_cardinality, but not both"""
+        if dimension.exact_cardinality is not None and (
+            dimension.minimum_cardinality is not None or dimension.maximum_cardinality is not None
+        ):
+            raise ValidationError(
+                "Can only specify EITHER exact_cardinality OR minimum/maximum cardinality, " f"got: {dimension}"
```

Comment:
```suggestion
                "Can only specify EITHER exact_cardinality OR minimum/maximum cardinality, " 
                f"got: {dimension}"
```

Reply:
i think black will probably want to put this back on one line, but if it does i'll combine the strings